src/conf.rs

```diff
@@ -19,6 +23,99 @@ pub enum BoomConfigError {
     ConfigFileNotFound,
     #[error("missing key in config")]
     MissingKeyError,
+    #[error("environment variable expansion error")]
+    EnvExpansionError(#[from] ExpandError),
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum ExpandError {
+    #[error("Missing environment variable '{var_name}' for placeholder '{placeholder}'")]
+    MissingVariable {
+        var_name: String,
+        placeholder: String,
+    },
+}
+
+/// Loads environment variables from a .env file if it exists.
+/// This function should be called early in the application startup,
+/// typically before any configuration loading.
+///
+/// The function looks for .env files in this order:
+/// 1. .env in the current working directory
+/// 2. .env in the parent directory (useful when running from subdirs)
+/// 3. If none found, continues without error (env vars may be set by system)
+pub fn load_dotenv() {
```

Comment:
I think this PR makes the duplication issue between the main app and the API even more obvious then before, we should really get to this soon-ish...